[PowerToys]File actions menu docs (#5137)
MicrosoftDocs/windows-uwp: hub/powertoys/file-actions-menu.md

---
title: PowerToys File Actions Menu

> description: A per shortcut invokable menu for certain file actions

Reviewer: ```suggestion
description: A per-shortcut, invokable context menu containing a set of file actions.
```

ms.date: 09/22/2024
ms.topic: article
ms.localizationpriority: medium
no-loc: [PowerToys, Win]
---

# File Actions menu

> ![File Actions menu](../images/PowerToys-FileActionsMenu-HashMenu.png)

Reviewer: ```suggestion
![A screenshot of the File Actions menu](../images/PowerToys-FileActionsMenu-HashMenu.png)
```

> The PowerToys File Action Menu is a menu, which is invoked by a shortcut and allows you to perform certain actions on your selected files.

Reviewer: ```suggestion
The PowerToys File Action Menu is a menu, which is invoked by a shortcut and allows you to perform a set of actions on the selected file(s).
```

| Merge contents | Merges the contents of the selected files into a new file | 2 |
| Copy content to clipboard | Copies the content of the selected file into the clipboard. If it is a directory the tree of the directory will be copied into the clipboard. | 1 |
| Collapse folder structure | Collapses the whole folder structure inside the selected folder | 1, 6 |
| Copy file content | Copies the file content into the clipboard in a certain way.<br/>**As plaintext**: Copies the whole file as normal text.<br/>**As base64 data link**: Copies the file content as a data url with base64 encoded content. For more information you can read the [specification](https://www.rfc-editor.org/rfc/rfc2397).<br/>**C escaped**: Copies the file content as a C escaped string (`\` → `\\`, newline → `\n\r`, etc.).<br/>**XML escaped**: Copies the content of the file with XML escape sequences (ex. `<` → `&#x3C;`)<br/>**C byte escaped**: Copies the file content, where every character is replaced with the ascii byte sequence (ex. `<` → `\0x3C`).<br/>**URI escaped**: Copies the file content ready to use for an URI (ex. `<` → `%3C`), | 1 |

## Supported hash types

> [!WARNING]

> > Certain hash types are nowadays considered unsafe (like md5 or SHA1). Please inform yourself about the insecurities of the hash types you use.

Reviewer: I would have pointed to a page on Learn for hash info, but some of those pages could use some updates.
```suggestion
> Today, some hash types ((like md5 or SHA1)) are considered unsafe. It's important to understand the insecurities of the hash types you use. Learn more about [Secure Hash Algorithms on Wikipedia](https://wikipedia.org/wiki/Secure_Hash_Algorithms).
```

Following types of hashes are supported.

- CRC32 Decimal
- CRC32 Hex
- CRC64 Decimal
- CRC64 Hex
- MD5
- SHA1
- SHA256
- SHA384
- SHA512
- SHA3-256
- SHA3-384
- SHA3-512

## Resolve shortcut

> Some actions allow the option to resolve shortcuts. This will target the files that the shortcuts point to instead of the `.ink`-file itsself.

Reviewer: The include file with PowerToys install info should be added to the end of every page. We get a lot of feedback from people who don't understand that each of these utilities needs to be installed with PowerToys.
```suggestion
Some actions allow the option to resolve shortcuts. This will target the files that the shortcuts point to instead of the `.ink`-file itsself.

[!INCLUDE [install-powertoys.md](../includes/install-powertoys.md)]
```
